code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,477 @@
+	active = FALSE
+	priority_announce("WARNING - The crystal has reached critical instability point. ZK-Event inbound, please do not panic, anyone who panics will \
+						be terminated on the spot. Have a nice day", "Alert")
+	sound_to_playing_players('sound/machines/alarm.ogg')
+	sleep(15 SECONDS)
+
+	var/list/spawners = list()
+	for(var/obj/effect/landmark/event_spawn/temp in GLOB.generic_event_spawns)
+		if(QDELETED(temp))
+			continue
+		if(is_station_level(temp.loc.z))
+			spawners += temp
+
+	if(!spawners.len)
+		message_admins("No landmarks on the station, aborting")
+		return MAP_ERROR
+
+	var/obj/spawner = pick_n_take(spawners)
+	var/pick_portal = pickweight(GLOB.waves["huge wave"])
+	for(var/i = 0, i < 10, i++)
+		new pick_portal(spawner.loc)
+	explosion(src, 7, 10, 25, 25)
+	is_zk = TRUE
+	qdel(src)
+
+/obj/machinery/destabilized_crystal/attackby(obj/item/W, mob/living/user, params)
+	if(!istype(W) || (W.item_flags & ABSTRACT) || !istype(user))
+		return
+	if(istype(W, /obj/item/crystal_stabilizer))
+		var/obj/item/crystal_stabilizer/injector = W
+		if(!injector.filled)
+			to_chat(user, "<span class='notice'>\The [W] is empty!</span>")
+			return
+		to_chat(user, "<span class='notice'>You carefully begin inject \the [src] with \the [W]... please don't move untill all the steps are finished</span>")
+		if(W.use_tool(src, user, 5 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>Seems that \the [src] internal resonance is fading with the fluid!</span>")
+			playsound(get_turf(src), 'sound/effects/supermatter.ogg', 35, TRUE)
+			if(W.use_tool(src, user, 6.5 SECONDS, volume=100))
+				to_chat(user, "<span class='notice'>The [src] is reacting violently with the fluid!</span>")
+				src.fire_nuclear_particle()
+				radiation_pulse(src, 2500, 6)
+				if(W.use_tool(src, user, 7.5 SECONDS, volume=100))
+					to_chat(user, "<span class='notice'>The [src] has been restored and restabilized!</span>")
+					playsound(get_turf(src), 'sound/effects/supermatter.ogg', 35, TRUE)
+					injector.filled = FALSE
+					active = FALSE
+					restore()
+
+///Restore the Destabilized Crystal as it was before
+/obj/machinery/destabilized_crystal/proc/restore()
+	priority_announce("The Crystal has been restored and is now stable again, your sector of space is now safe from the ZK-Lambda-Class Scenario, \
+						kill the remaining crystal monsters and go back to work")
+	sound_to_playing_players('sound/misc/notice2.ogg')
+	var/turf/T = get_turf(src)
+	new/obj/machinery/power/supermatter_crystal(T)
+	for(var/Portal in GLOB.crystal_portals)
+		qdel(Portal)
+	qdel(src)
+
+/*
+This section is for the crystal stabilizer item
+*/
+/obj/item/crystal_stabilizer
+	name = "Supermatter Stabilizer"
+	desc = "Used when the Supermatter Matrix is starting to reach the destruction point."
+	icon = 'icons/obj/supermatter.dmi'
+	icon_state = "stabilizer"
+	slot_flags = ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_HUGE
+	///The stabilizer is one use only
+	var/filled = FALSE
```

Comment:
maybe tell people that it is loaded over an examine message.

Reply:
will do